GoogleSignIn/Sources/GIDSignIn.m

```diff
@@ -541,6 +541,43 @@ - (void)authenticateWithOptions:(GIDSignInInternalOptions *)options {
   [self addCompletionCallback:authFlow];
 }
 
+// Restores the current user with the authorization object from the keychain.
+- (void)restoreCurrentUserWithAuthState:(OIDAuthState *)authState {
+  if (!authState) {
+    return;
+  }
+
+  OIDIDToken *idToken =
+      [[OIDIDToken alloc] initWithIDTokenString:authState.lastTokenResponse.idToken];
+  GIDProfileData *profileData = [self profileDataWithIDToken:idToken];
+
+  GIDGoogleUser *user = [[GIDGoogleUser alloc] initWithAuthState:authState profileData:profileData];
+  [self setCurrentUserWithKVO:user];
+}
+
+// Generates user profile from OIDIDToken.
+- (GIDProfileData *)profileDataWithIDToken:(OIDIDToken *)idToken {
+  if (!idToken || !idToken.claims[kBasicProfilePictureKey] ||
+      !idToken.claims[kBasicProfileNameKey] || !idToken.claims[kBasicProfileGivenNameKey] ||
+      !idToken.claims[kBasicProfileFamilyNameKey]) {
+    return nil;
+  } else {
+    return [[GIDProfileData alloc]
+        initWithEmail:idToken.claims[kBasicProfileEmailKey]
+                 name:idToken.claims[kBasicProfileNameKey]
+            givenName:idToken.claims[kBasicProfileGivenNameKey]
+           familyName:idToken.claims[kBasicProfileFamilyNameKey]
+             imageURL:[NSURL URLWithString:idToken.claims[kBasicProfilePictureKey]]];
+  }
```

Comment:
You can drop "else" here. It's more common to do:

```
if (/* error condition */) {
  return;
}
// handle success case
```

Reply:
Done.